static/js/components/EducationDialog.js

```diff
@@ -23,11 +26,16 @@ export default class EducationDialog extends ProfileFormFields {
   }
   educationLevelLabels: Object;
 
-  static propTypes = {
-    open:           React.PropTypes.bool,
-    onClose:        React.PropTypes.func,
-    onSave:         React.PropTypes.func,
-    showLevelForm:  React.PropTypes.bool,
+  props: {
+    ui:                           UIState,
+    setEducationDialogVisibility: () => void,
+    setEducationDegreeLevel:      () => void,
+    setEducationDialogIndex:      () => void,
+    clearProfileEdit:             () => void,
+    saveProfile:                  BoundSaveProfile,
```

Comment:
you've added some types here, which is awesome, but a few of them have sort of jarring names (`BoundSaveProfile`, `BoundUpdateProfile`, `SetWorkHistoryEdit` etc). let's say i'm reading through these props and trying to get a sense of what everything is.

> what is this 'profile' prop? oh, it's a Profile. what is the 'showLevelForm' variable supposed to be? oh, it's a 'boolean'. what is 'saveProfile'? it's a... BoundSaveProfile. wat?

i think that type should be called `ProfileSaveFunction` or something like that. "what am i going to be getting in the 'saveProfile' prop? ah, i'll be getting a 'ProfileSaveFunction'." 

that is, of course, assuming that we need a declared type. my examples above (`BoundSaveProfile`, `BoundUpdateProfile`, `SetWorkHistoryEdit`) seem to be single functions, ie: there aren't several functions that conform to that 'type'. i'll defer to you to decide whether we need to keep these around as types